tests/unit/mainwindow/test_tabwidget.py

```diff
@@ -52,3 +52,21 @@ def test_small_icon_doesnt_crash(self, widget, qtbot, fake_web_tab):
 
         with qtbot.waitExposed(widget):
             widget.show()
+
+    def test_update_tab_titles_benchmark(self, benchmark, widget,
+                                         qtbot, fake_web_tab):
+        """Benchmark for update_tab_titles."""
+        widget.addTab(fake_web_tab(), 'foobar')
+        widget.addTab(fake_web_tab(), 'foobar2')
+        widget.addTab(fake_web_tab(), 'foobar3')
+        widget.addTab(fake_web_tab(), 'foobar4')
+
+        with qtbot.waitExposed(widget):
+            widget.show()
+
+        def bench():
+            for _a in range(1000):
+                # pylint: disable=protected-access
```

Comment:
`protected-access` is [automatically disabled](https://github.com/qutebrowser/qutebrowser/blob/43aa7423abb931ddcbccc51411f6a8a16bf056b5/scripts/dev/run_pylint_on_tests.py#L55) during tests already.